bedita-app/models/behaviors/cacheable.php

```diff
@@ -103,30 +103,35 @@ public function setObjectsToClean(&$model, $objectId, array $excludeIds = array(
      */
     public function getObjectsToCleanById(&$model, $objectId, array $excludeIds = array()) {
         // get parents to clean
-        $tree = ClassRegistry::init('Tree');
-        $treeConditions = array('id' => $objectId);
-        if (!empty($excludeIds)) {
-            $treeConditions['NOT'] = array('parent_id' => $excludeIds);
+        $parents = array();
+        // if destination data, use it instead of calling a tree->find
+        if (!empty($model->data['BEObject']['destination'])) {
```

Comment:
parents in `destination` should be correctly handled in `beforeSave()` I think.

```php
$treeIds = (!empty($data['destination'])) ? $data['destination'] : array();
$excludeIdsFromQuery = array_merge($treeIds, $relatedIds);
// prepare objects to clean
if (!empty($data['id'])) {
    $this->setObjectsToClean($model, $data['id'], $excludeIdsFromQuery);
}
$this->addObjectsToClean($model, $excludeIdsFromQuery);
```

Before saving an object the parents of the object we're going to save and all those in `destination` array are collects to clear them cache later.

Reply:
yes... but the problem is that a find is always performed (in 3-corylus), and sometimes data is not yet in tree. that's why we use here this patch: do not use tree->find, if parents are already available